sync/src/synchronizer/mod.rs

```diff
@@ -207,21 +213,20 @@ impl Synchronizer {
     pub fn eviction(&self, nc: &dyn CKBProtocolContext) {
         let mut peer_states = self.peers().state.write();
         let active_chain = self.shared.active_chain();
-        let is_initial_header_sync = self.shared.state().is_initial_header_sync();
         let mut eviction = Vec::new();
+        let better_tip = self.better_tip();
         for (peer, state) in peer_states.iter_mut() {
             let now = unix_time_as_millis();
 
-            // headers_sync_timeout
-            if let Some(timeout) = state.headers_sync_timeout {
-                if is_initial_header_sync {
-                    if now > timeout && !state.disconnect {
+            if let Some(ref mut process) = state.headers_sync_process {
+                if let Some(is_timeout) = process.is_timeout(&better_tip, now) {
```

Comment:
As I mentioned in the comment on `HEADERS_DOWNLOAD_TIMEOUT_BASE`, this is not a timeout on downloaded headers, but indeed the timeout on the tip. So the timeout should also use active chain tip to check.

Reply:
I think there still need two timeouts or two mechanisms:
1. In IBD, I known some blocks were already existed, so I hoped I could download headers of them more quickly（far greater than `1 minute-in-blockchain/minute-in-reality`).
  So, I can reach the tip in a reasonable time.
2. After IBD, I was on the tip, so I only expected I could download headers not too slow. So slightly slower than `1 minute-in-blockchain/minute-in-reality` is acceptable.